### DISCUSSION TITLE
Renders don't show up in the gameRenders folder (Ubuntu 24.04 LTS x64)

### DISCUSSION THREAD

**Discussion by u0-a63** [JordanSantiagoYT/FNF-JS-Engine#425] [General]
### Describe your bug here. Mind you, if you are modding, please try testing your bug in a clean version of the engine instead, this ensures the bug is actually caused by the engine itself. Also, if you're using an older version of JSE, please try the latest action build in the Actions tab or the latest release build. The issue has probably been fixed since that version. Oh yeah, and also be sure to check the pinned 'JS Engine: Known Issues' issue, to double check if your issue hasn't already been found!

When I use Rendering Mode, NOTHING appears in the renders folder.
Yes, I do have the ffmpeg package installed. Do I have to do something else to get this to work?

### Command Prompt/Terminal/Crash logs (if existing)

```bash
source/TitleState.hx:143: checking for update
source/TitleState.hx:156: version online: 1.27.2, your version: 1.27.2
source/TitleState.hx:163: the versions match!
source/LoadingState.hx:162: Setting asset folder to shared
source/FreeplayState.hx:610: talking
source/FreeplayState.hx:624: CURRENT WEEK: funnyweek
source/LoadingState.hx:162: Setting asset folder to shared
source/PlayState.hx:7824: "FFmpeg.exe" not found! (Is it in the same folder as the JS Engine exe?)
source/PlayState.hx:2070: Anim speed: 1
source/Conductor.hx:156: new BPM map BUDDY [{ stepTime => 0, stepCrochet => 21.4285714285714, bpm => 700, songTime => 0 }]
source/Paths.hx:121: NOTE_assets
source/FunkinLua.hx:2922: initializing haxe interp for: mods/funnysongs/scripts/dyed.lua
source/FunkinLua.hx:114: lua file loaded succesfully:mods/funnysongs/scripts/dyed.lua
source/PlayState.hx:1394: Loading chart...
source/Paths.hx:725: oh no its returning null NOOOO (assets/songs/talking/Voices.ogg)
source/PlayState.hx:3412: Done! 1537 notes were loaded in 0.022446870803833 seconds.
source/PlayState.hx:5424: cached gold ratings
source/PlayState.hx:2070: Anim speed: 1
source/PlayState.hx:5329: WENT BACK TO FREEPLAY??
```

### Have you identified any steps to reproduce the bug? If so, please describe them below in as much detail as possible. Use images if possible.

```bash
Try to render a song on the latest Ubuntu version (24.04 LTS as of writing this issue), it doesn't work even if you install the ffmpeg package.
```

### Are you modding a build from source or with Lua?

Lua

### What is your build target?

Linux

### Did you edit anything in this build? If so, mention or summarize your changes.

No, I only added a mod.

### If you use 1.12.0 or earlier, did you have Optimized Chart Loading turned on?

None

### Did you check for any similar issues to what you're reporting? Check, then come back here. If there is a similar issue, then do not report the issue, otherwise it will be marked as a duplicate.

Yes

**Comment by JordanSantiagoYT** [comment]
...

it says "Is ffmpeg.exe in the same folder as the JS Engine exe?"

**Comment by u0-a63** [comment]
Yes, I know, **but Ubuntu doesn't use .exe files!**

**Comment by u0-a63** [comment]
Also, stupid thing to ask, but, can you fix the camera shaking when you run the game at 30FPS? Watch the video file I attached (the pixeldrain link) and you'll see what I mean.
https://pixeldrain.com/u/empxj4qi

**Comment by ACoolioDude** [comment]
> ...
> 
> it says "Is ffmpeg.exe in the same folder as the JS Engine exe?"

Bruh, blud doesn't know about Linux execution files 💀

**Comment by JordanSantiagoYT** [comment]
> Yes, I know, **but Ubuntu doesn't use .exe files!**

my bad, i forgot to make the trace thing trace only "'ffmpeg not found!" instead of "ffmpeg.exe not found!"

did you put the ffmpeg file in the same location as JSEngine?

**Comment by JordanSantiagoYT** [comment]
> Also, stupid thing to ask, but, can you fix the camera shaking when you run the game at 30FPS? Watch the video file I attached (the pixeldrain link) and you'll see what I mean. https://pixeldrain.com/u/empxj4qi

i am aware of this issue, but there's nothing i can do to fix it because the last time i tried there were issues with the camera following the players too slowly.

**Comment by u0-a63** [comment]
Even on Windows, there's a problem with Rendering Mode. There's no sound in the finished .mp4 file. Can you fix this?

**Comment by ACoolioDude** [comment]
> Even on Windows, there's a problem with Rendering Mode. There's no sound in the finished .mp4 file. Can you fix this?

Rendering Mode should be removed from game when you can literally use OBS for that. I don't care how much it will give a huge CPU usage and lag, just configure the settings and that's all

**Comment by JordanSantiagoYT** [comment]
> Even on Windows, there's a problem with Rendering Mode. There's no sound in the finished .mp4 file. Can you fix this?

..you're supposed to use a video editor to edit it in

**Comment by JordanSantiagoYT** [comment]
> Rendering Mode should be removed from game when you can literally use OBS for that. I don't care how much it will give a huge CPU usage and lag, just configure the settings and that's all

the point of rendering mode is to render charts without lag, and without having to wait a whole hour just to record 1 CHART (as is the case with [this chart](https://www.youtube.com/watch?v=OUT5n0Vz4w0))
![image](https://github.com/JordanSantiagoYT/FNF-PsychEngine-NoBotplayLag/assets/61796448/7cf793e1-6e57-4ff8-b9b4-ad2a03938205)

**Comment by ACoolioDude** [comment]
> > Rendering Mode should be removed from game when you can literally use OBS for that. I don't care how much it will give a huge CPU usage and lag, just configure the settings and that's all
> 
> the point of rendering mode is to render charts without lag, and without having to wait a whole hour just to record 1 CHART (as is the case with [this chart](https://www.youtube.com/watch?v=OUT5n0Vz4w0)) ![image](https://private-user-images.githubusercontent.com/61796448/345093635-7cf793e1-6e57-4ff8-b9b4-ad2a03938205.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5NzAzODUsIm5iZiI6MTcxOTk3MDA4NSwicGF0aCI6Ii82MTc5NjQ0OC8zNDUwOTM2MzUtN2NmNzkzZTEtNmU1Ny00ZmY4LWI5YjQtYWQyYTAzOTM4MjA1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAzVDAxMjgwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg0OWZmMjhjZDQzNmMyMDAzMzdmNmM2OWU2OWY3ODcyZmJjMDBlM2UwYTA2YjhhNWEwMTgyMjcwNTc1YjE1YzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.M4P8SwRjyHcIx8PXC06Ri6DqIQ7k2sJrIBI30lvSsVQ)

Doesn't matter. Buy a better CPU 😐. Work harder for money to buy it. I already got 60€ in my country since i worked harder for money. You just think this will change FNF forever, but no, it will no. Haxe server also hates rendering mode because it's useless thing i ever saw. You know Jordan, maybe i will not have the best CPU on my laptop, but i will still choose OBS for recording. Actually, you know what? I don't really care, i am just gonna leave you whatever you doing with your fork, i am porting new stuff from FNF 0.3+ to my modified Psych fork, so bye

**Comment by ACoolioDude** [comment]
> > Even on Windows, there's a problem with Rendering Mode. There's no sound in the finished .mp4 file. Can you fix this?
> 
> ..you're supposed to use a video editor to edit it in

Awesome, now how people will get a sound Jordan? You can code everything, right? If is true, then make a function to play audio too

**Comment by JordanSantiagoYT** [comment]
> > > Even on Windows, there's a problem with Rendering Mode. There's no sound in the finished .mp4 file. Can you fix this?
> > 
> > 
> > ..you're supposed to use a video editor to edit it in
> 
> Awesome, now how people will get a sound Jordan? You can code everything, right? If is true, then make a function to play audio too

the reason i don't do that is because it doesn't work as shown in this video

https://github.com/JordanSantiagoYT/FNF-PsychEngine-NoBotplayLag/assets/61796448/28e053a2-3963-4a5d-af3f-c509a1af6b6d

**Comment by JordanSantiagoYT** [comment]
if you're willing to fix this code to make it so that it automatically adds the song audio, then be my guest

		var instPath = Paths.songPath(SONG.song, CoolUtil.difficultyString(), false);
		var voicesPath = Paths.songPath(SONG.song, CoolUtil.difficultyString(), true);

		var audioDelay:Float = !skipCountdown ? Conductor.crochet * 5 : 0;

                //Feel free to remove these traces, these just verify that they work
		trace (instPath);
		trace (SONG.needsVoices ? voicesPath : 'doesnt need voices');

		if (!SONG.needsVoices)
			process = new sys.io.Process('ffmpeg', [
				'-f', 'rawvideo',
				'-pix_fmt', 'rgba',
				'-s', lime.app.Application.current.window.width + 'x' + lime.app.Application.current.window.height,
				'-r', Std.string(targetFPS),
				'-i', '-',
				'-i', Std.string(instPath),
				'-filter_complex', '"[1:a]adelay=${audioDelay}|${audioDelay}[aout]"',
				'-map', '0:v', '-map', '"[aout]"',
				'assets/gameRenders/' + Paths.formatToSongPath(SONG.song) + '.mp4'
			]);
		else
		//code for if the song needs voices

and this is for songPath

```
static public function songPath(song:String, ?difficulty:String = '', ?voices:Bool = false):Any
	{
		var instOrVoices:String = (voices ? 'Voices' : 'Inst');
		var formattedDifficulty:String = formatToSongPath(difficulty);
		if (difficulty.contains(' ')) difficulty = formattedDifficulty;
		#if html5
		return 'songs:assets/songs/${formatToSongPath(song)}/${instOrVoices}.$SOUND_EXT';
		#else
		if (difficulty != null)
		{
			var songKey:String = '${formatToSongPath(song)}/${instOrVoices}-$difficulty';
			if (FileSystem.exists(Paths.modFolders('songs/' + songKey + '.$SOUND_EXT')) || FileSystem.exists('assets/songs/' + songKey + '.$SOUND_EXT')) 
			{
				return songKey + '.$SOUND_EXT';
			}
		}
		var songKey:String = '${formatToSongPath(song)}/${instOrVoices}.$SOUND_EXT';
		if (FileSystem.exists('assets/songs/$songKey')) return 'assets/songs/$songKey';
		if (FileSystem.exists(Paths.modFolders('songs/' + songKey + '.$SOUND_EXT')))
			return Paths.modFolders('songs/' + songKey + '.$SOUND_EXT');

		else return null;
		#end
	}
```

**Comment by cabfile** [comment]
> > > Rendering Mode should be removed from game when you can literally use OBS for that. I don't care how much it will give a huge CPU usage and lag, just configure the settings and that's all
> > 
> > 
> > the point of rendering mode is to render charts without lag, and without having to wait a whole hour just to record 1 CHART (as is the case with [this chart](https://www.youtube.com/watch?v=OUT5n0Vz4w0)) ![image](https://private-user-images.githubusercontent.com/61796448/345093635-7cf793e1-6e57-4ff8-b9b4-ad2a03938205.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5NzAzODUsIm5iZiI6MTcxOTk3MDA4NSwicGF0aCI6Ii82MTc5NjQ0OC8zNDUwOTM2MzUtN2NmNzkzZTEtNmU1Ny00ZmY4LWI5YjQtYWQyYTAzOTM4MjA1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAzVDAxMjgwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg0OWZmMjhjZDQzNmMyMDAzMzdmNmM2OWU2OWY3ODcyZmJjMDBlM2UwYTA2YjhhNWEwMTgyMjcwNTc1YjE1YzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.M4P8SwRjyHcIx8PXC06Ri6DqIQ7k2sJrIBI30lvSsVQ)
> 
> Doesn't matter. Buy a better CPU 😐. Work harder for money to buy it. I already got 60€ in my country since i worked harder for money. You just think this will change FNF forever, but no, it will no. Haxe server also hates rendering mode because it's useless thing i ever saw. You know Jordan, maybe i will not have the best CPU on my laptop, but i will still choose OBS for recording. Actually, you know what? I don't really care, i am just gonna leave you whatever you doing with your fork, i am porting new stuff from FNF 0.3+ to my modified Psych fork, so bye

what the fuck are you talking about??

not everyone with low-end pcs have money and not everyone have the time and patience to earn it.. this is the same as saying "erm lol skill issue" to a person with a high-end pc struggling to run a clearly poorly optimized game (not saying js engine is poorly optimized though)

rendering mode has a point as well in that it provides lag-less recordings. if you use obs you'll have to use a really slow playback rate and then wait a million years for a song to finish, and then speed that recording up in a video editor to get the same effect. this will take an unnecessary amount of disk space (and time!) as well.

oh and good luck with your fork

**Comment by cabfile** [comment]
> > Even on Windows, there's a problem with Rendering Mode. There's no sound in the finished .mp4 file. Can you fix this?
> 
> ..you're supposed to use a video editor to edit it in

my original lua implementation already automatically adds music in

**Comment by JordanSantiagoYT** [comment]
cabfile has a point with this.

i'm not going to get the latest intel processor just so i can render miscreancy without rendering mode or playback rates, not only is it a complete waste of money but it's absolutely stupid

and not everyone with a low-end pc is going to have the time and patience to earn a better pc. THAT COULD TAKE YEARS FOR GODS SAKE.

the reason i use rendering mode for all my charts from now instead of using OBS is because of 2 things:
1. it will occasionally drop to 59 FPS, and for some reason capcut converts that to 30 FPS
2. it gets annoying when you need to use a slower playback rate and then wait much longer than you would have to if you used rendering mode. then you have to speed it up in the video editor. rendering mode saves disk space.

**Comment by ACoolioDude** [comment]
Well, womp womp you two. You just think people are just lazy to edit OBS settings to get something

**Comment by JordanSantiagoYT** [comment]
> Well, womp womp you two. You just think people are just lazy to edit OBS settings to get something

configuring either obs settings or rendering mode settings takes only a few minutes, so i wouldnt call people who dont use obs lazy

let people use whichever one they prefer. i'm not saying you HAVE to use rendering mode, but you're basically saying that people HAVE to use OBS because you think it's superior.

![image](https://github.com/JordanSantiagoYT/FNF-PsychEngine-NoBotplayLag/assets/61796448/6fa9d1cf-59ed-4261-bd44-9dc797462436)
also, can you remove this tip from your tipsText? i don't talk bad about your engine, so how is it that you can talk like that about mine?

**Comment by moxie-coder** [comment]
> > Well, womp womp you two. You just think people are just lazy to edit OBS settings to get something
> 
> configuring either obs settings or rendering mode settings takes only a few minutes, so i wouldnt call people who dont use obs lazy
> 
> let people use whichever one they prefer. i'm not saying you HAVE to use rendering mode, but you're basically saying that people HAVE to use OBS because you think it's superior.
> 
> ![image](https://private-user-images.githubusercontent.com/61796448/345641544-6fa9d1cf-59ed-4261-bd44-9dc797462436.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwNDc3OTcsIm5iZiI6MTcyMDA0NzQ5NywicGF0aCI6Ii82MTc5NjQ0OC8zNDU2NDE1NDQtNmZhOWQxY2YtNTllZC00MjYxLWJkNDQtOWRjNzk3NDYyNDM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAzVDIyNTgxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0MDc5ZThkM2Q4MjMwMzYyYmYxOWYzZWQzZTQ3MmFiNDcwNzNmYjY4ODQxYjIzMTNhODUxZDQzMjQ2OWUxYmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Y4a1ZfWL3jLQXrP-zw_DinyPGx6ab5SFwr_DD9iaNA8) also, can you remove this tip from your tipsText? i don't talk bad about your engine, so how is it that you can talk like that about mine?

![IMG_9707](https://github.com/JordanSantiagoYT/FNF-PsychEngine-NoBotplayLag/assets/124418090/38f90db0-a73a-4178-b295-99f5401b51b8)
not really an valid argument honestly because you put this here

**Comment by JordanSantiagoYT** [comment]
> ![IMG_9707](https://private-user-images.githubusercontent.com/124418090/345642011-38f90db0-a73a-4178-b295-99f5401b51b8.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwNDc4MjYsIm5iZiI6MTcyMDA0NzUyNiwicGF0aCI6Ii8xMjQ0MTgwOTAvMzQ1NjQyMDExLTM4ZjkwZGIwLWE3M2EtNDE3OC1iMjk1LTk5ZjU0MDFiNTFiOC5qcGVnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MDNUMjI1ODQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWM2MzYzNzNmOGQwYzhjNWY4NGE0NGU4OTRmZjc4MDBkM2VmOTc5YzI5NGY4Y2Y1NWIxNzVhNTMzOGU5NWQwZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.LqFHMaoQLkbgZZF9xLVtQk6sqhQMRMKrxwjbJDB_S54) not really an valid argument honestly because you put this here

i only put that there in response to finding out that stefan put that tip there.

if he removes the tip, i will remove the tip in this picture.

**Reply by ACoolioDude** [reply]
> > ![IMG_9707](https://private-user-images.githubusercontent.com/124418090/345642011-38f90db0-a73a-4178-b295-99f5401b51b8.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwNDc4MjYsIm5iZiI6MTcyMDA0NzUyNiwicGF0aCI6Ii8xMjQ0MTgwOTAvMzQ1NjQyMDExLTM4ZjkwZGIwLWE3M2EtNDE3OC1iMjk1LTk5ZjU0MDFiNTFiOC5qcGVnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MDNUMjI1ODQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWM2MzYzNzNmOGQwYzhjNWY4NGE0NGU4OTRmZjc4MDBkM2VmOTc5YzI5NGY4Y2Y1NWIxNzVhNTMzOGU5NWQwZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.LqFHMaoQLkbgZZF9xLVtQk6sqhQMRMKrxwjbJDB_S54) not really an valid argument honestly because you put this here
> 
> i only put that there in response to finding out that stefan put that tip there.
> 
> if he removes the tip, i will remove the tip in this picture.

No

**Comment by moxie-coder** [comment]
> > ![IMG_9707](https://private-user-images.githubusercontent.com/124418090/345642011-38f90db0-a73a-4178-b295-99f5401b51b8.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwNDc4MjYsIm5iZiI6MTcyMDA0NzUyNiwicGF0aCI6Ii8xMjQ0MTgwOTAvMzQ1NjQyMDExLTM4ZjkwZGIwLWE3M2EtNDE3OC1iMjk1LTk5ZjU0MDFiNTFiOC5qcGVnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MDNUMjI1ODQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWM2MzYzNzNmOGQwYzhjNWY4NGE0NGU4OTRmZjc4MDBkM2VmOTc5YzI5NGY4Y2Y1NWIxNzVhNTMzOGU5NWQwZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.LqFHMaoQLkbgZZF9xLVtQk6sqhQMRMKrxwjbJDB_S54) not really an valid argument honestly because you put this here
> 
> i only put that there in response to finding out that stefan put that tip there.
> 
> if he removes the tip, i will remove the tip in this picture.

okay, I’ll be honest people hate this engine because of the lack of proper optimization, memory leaks, and how bloated it is

**Comment by moxie-coder** [comment]
a lot of people hate it for that reason, even with gpu caching I saw the ram in week 5 peak as far as 700 mb of ram, that’s far more ram then it should be using

**Comment by moxie-coder** [comment]
and for those kinds of reason, I don’t think I can recommend this engine for regular use, other then spam tracks but that’s about it, actually I don’t think I can recommend this engine in general because there’s so much bloat in this engine, and it’s caused some issues for me

**Comment by moxie-coder** [comment]
and there’s definitely some bugs I’ve encountered for a while that are pretty minor, but memory leaks in this engine is definitely a problem, and the lack of optimization in general because gpu caching didn’t really help too much, so it’s all kind of fucked

**Comment by moxie-coder** [comment]
like, the whole point of psych engine was to optimize the game so it could run on older pcs compared to many engines like Kade engine, and the easy modding, this from what I can see definitely is not made for older pcs because if it crashes for Stefan on his old 1.10GHZ cpu, then I don’t wanna know how bad it would be on my old laptops, and also it starts to slow down and lag overtime when using this engine too, unlike psych

**Comment by moxie-coder** [comment]
this is just my critique, but like, most of these problems do stand though, and I do wanna help fix this engine and the problems it has

**Comment by moxie-coder** [comment]
you don't gotta listen to me @JordanSantiagoYT though, since I was just trying to give constructive criticism, and feedback

**Comment by ACoolioDude** [comment]
> > Well, womp womp you two. You just think people are just lazy to edit OBS settings to get something
> 
> configuring either obs settings or rendering mode settings takes only a few minutes, so i wouldnt call people who dont use obs lazy
> 
> let people use whichever one they prefer. i'm not saying you HAVE to use rendering mode, but you're basically saying that people HAVE to use OBS because you think it's superior.
> 
> ![image](https://private-user-images.githubusercontent.com/61796448/345641544-6fa9d1cf-59ed-4261-bd44-9dc797462436.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwNTAyMzgsIm5iZiI6MTcyMDA0OTkzOCwicGF0aCI6Ii82MTc5NjQ0OC8zNDU2NDE1NDQtNmZhOWQxY2YtNTllZC00MjYxLWJkNDQtOWRjNzk3NDYyNDM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAzVDIzMzg1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0NmU5MjlhNDVmMDc1NWRhZWZlZjZjYTA1YmMyYWRkYTc5ZmY0YzMzM2Q5NDJmM2I5ZmFhNDkyZGFlYWZjOGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KhrNftQBLZg87z43wgFYt36r5vRTGL7D2uRyxc6QudA) also, can you remove this tip from your tipsText? i don't talk bad about your engine, so how is it that you can talk like that about mine?

Reason: population, many bugs with Hide HUD option

**Comment by cyklusiguess** [comment]
what the fuck just happened here dawg

**Comment by moxie-coder** [comment]
> what the fuck just happened here dawg

a lot, I’m turning this into a discussion

**Comment by cabfile** [comment]
i agree with moxie, js engine is bloated. (a million customizability settings...) when i finally decided to check out the engine, i was overwhelmed with the sheer amount of options. similar thing happened when i tried out denpaex.

js engine to me seems more like vlc, winamp or any other such media player, but for fnf songs specifically. it doesnt work as something you would carry around alongside a mod, like strident crisis' psych fork does.

i really would like this engine to become much, much more optimized. unfortunately most people who look at this engine either go "ew bambi" or "this code sucks", and in some rare occasions "this engine uses stolen code!!!!!!!". (well, if its stolen, why wont you bother rewriting it then, if you care so much?) as a result i can clearly see there's less than _5_ people more or less actively working on the engine.

**Reply by JordanSantiagoYT** [reply]
> "this engine uses stolen code!!!!!!!". (well, if its stolen, why wont you bother rewriting it then, if you care so much?)

LMA

**Reply by moxie-coder** [reply]
> i agree with moxie, js engine is bloated. (a million customizability settings...) when i finally decided to check out the engine, i was overwhelmed with the sheer amount of options. similar thing happened when i tried out denpaex.
> 
> js engine to me seems more like vlc, winamp or any other such media player, but for fnf songs specifically. it doesnt work as something you would carry around alongside a mod, like strident crisis' psych fork does.
> 
> i really would like this engine to become much, much more optimized. unfortunately most people who look at this engine either go "ew bambi" or "this code sucks", and in some rare occasions "this engine uses stolen code!!!!!!!". (well, if its stolen, why wont you bother rewriting it then, if you care so much?) as a result i can clearly see there's less than _5_ people more or less actively working on the engine.

Denpa engine is similar, and I've worked on it before, but it has actual organized code & proper optimization though, because it's worked on by people who *actually know* what they're doing, and this and Denpa engine both have similar goals, but not built the same way

**Comment by u0-a63** [comment]
woah, i never expected this to go from a simple issue to a full-on argument- holy...
anyways, you shouldn't have to edit in the audio. just don't mute the song audio in rendering mode, theres your solution...

**Reply by JordanSantiagoYT** [reply]
> woah, i never expected this to go from a simple issue to a full-on argument- holy... anyways, you shouldn't have to edit in the audio. just don't mute the song audio in rendering mode, theres your solution...

even then, the song audio will desync. + you still have to edit it in anyway as the final video still wont have sound
